### PR TITLE
Tests for source overriding (#2264)

### LINKED ISSUE #2264: Tests for source overriding
### Describe the feature
Sources in the root package can override the sources defined in dependencies. That's good! But we don't test that this is the case, and could easily regress. That is bad, especially because we want to document this behavior and tell users they can rely on it.

### Describe alternatives you've considered
No tests, probably break this in a future release, upset people.

### Additional context
N/A

### Who will this benefit?
Tests benefit everyone.

### PR DESCRIPTION
**Pull request by beckjake**
resolves #2264

### Description

- Make sure dependency sources can be selected from the root project
- Make sure the root project can override dependency sources

Is there more we want to assert on here?

### Checklist
 - [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [X] I have run this code in development and it appears to resolve the stated issue
 - [X] This PR includes tests, or tests are not required/relevant for this PR
 - [X] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,14 +5,17 @@
...
 
 ### Fixes
 - When a jinja value is undefined, give a helpful error instead of failing with cryptic "cannot pickle ParserMacroCapture" errors ([#2110](https://github.com/fishtown-analytics/dbt/issues/2110), [#2184](https://github.com/fishtown-analytics/dbt/pull/2184))
 - Added timeout to registry download call ([#2195](https://github.com/fishtown-analytics/dbt/issues/2195), [#2228](https://github.com/fishtown-analytics/dbt/pull/2228))
 - When a macro is called with invalid arguments, include the calling model in the output ([#2073](https://github.com/fishtown-analytics/dbt/issues/2073), [#2238](https://github.com/fishtown-analytics/dbt/pull/2238))
 - When a warn exception is not in a jinja do block, return an empty string instead of None ([#2222](https://github.com/fishtown-analytics/dbt/issues/2222), [#2259](https://github.com/fishtown-analytics/dbt/pull/2259))
 
+### Under the hood
+- Added more tests for source inhertiance ([#2264](https://github.com/fishtown-analytics/dbt/issues/2264), [#2291](https://github.com/fishtown-analytics/dbt/pull/2291))
```
Comment (drewbanin) [inline CHANGELOG.md:17]:
```suggestion
- Added more tests for source inheritance ([#2264](https://github.com/fishtown-analytics/dbt/issues/2264), [#2291](https://github.com/fishtown-analytics/dbt/pull/2291))
```

**Comment by beckjake** [comment]
>The list of columns for a source in local_models should override the list of columns in the dependency package for the same source. This would be most noticeable if there were column tests defined in the dependency package that should be overridden in the root project. This behavior might currently be poorly defined. I think this case, we'd just want to clobber the list of columns rather than deep-merge them.

This doesn't play well with how dbt behaves. Since the two sources both show up in the manifest as distinct entities (they're in different packages), test selection ends up selecting the tests for each source. I think the easiest fix to this to have test selection take care of this, but that feels all weird. If you want sources to truly overwrite, that's going to be quite the architectural change.

**Comment by beckjake** [comment]
>Tags from the root project for the source should override tags defined in the dependency package. I could buy that these tags should be extended, but I think it would probably be more consistent to clobber them instead. What do you think?

Similar to the above concern: What currently happens is both sources exist in the manifest and have their own lists of tags. So you kind of get both.

**Comment by beckjake** [comment]
>Source-level quoting configs from the root project should override source-level quoting configs from the dependency package.

Keeping with the theme here - currently each source config gets its own quoting config based on its package...

**Comment by beckjake** [comment]
It kind of feels like what you want here is for sources to _not_ have their package name in their unique ID, and therefore sources are sort of "communal" and each source table can override others in dependency packages... Is that dumb?

**Pull request merged**